lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -594,6 +594,64 @@ class EmitterBase {
                                  std::function<void(Value)> operandEmitter,
                                  ArrayAttr symAttrs, ModuleNameManager &names);
 
+  /// Emit the value of a StringAttr as one or more Verilog "one-line" comments
+  /// ("//").  Break the comment to respect the emittedLineLength and trim
+  /// whitespace after a line break.  Do nothing if the StringAttr is null or
+  /// the value is empty.
+  void emitComment(StringAttr comment) {
+    if (!comment)
+      return;
+
+    auto ref = comment.getValue();
+    if (ref.empty())
+      return;
+
+    // Set a line length for the comment.  Subtract off the leading comment and
+    // space ("// ") as well as the current indent level to simplify later
+    // arithmetic.  Ensure that this line length doesn't go below zero.
+    auto lineLength = state.options.emittedLineLength - state.currentIndent - 3;
+    if (lineLength > state.options.emittedLineLength)
+      lineLength = 0;
+
+    // Emit the comment and break lines.
```

Comment:
I'd recommend allowing \n in the comment string to indicate a forced line break.  This would allow adding a comment line: "Synoposis: this module does important stuff.\n\nDetails of this module include ........" or whatever.  Each line should get a `//` before it.

Does this handle escaping of weird characters correctly?

Reply:
Good thinking here, too. I had forgotten about this, even though Grand Central uses it. 🤦  User-specified breaking is super useful if you have an itemized list, table, diagram, etc. We may want to add an option here for "user-managed line breaking" to better enable the diagram use case.